src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/ElasticsearchQueryHelper.java

```diff
@@ -0,0 +1,166 @@
+import uk.gov.hmcts.ccd.domain.service.aggregated.AuthorisedGetCaseTypeOperation;
+import uk.gov.hmcts.ccd.domain.service.aggregated.GetCaseTypeOperation;
+import uk.gov.hmcts.ccd.domain.service.aggregated.SearchQueryOperation;
+import uk.gov.hmcts.ccd.domain.service.common.ObjectMapperService;
+import uk.gov.hmcts.ccd.endpoint.exceptions.BadSearchRequest;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.ccd.domain.service.common.AccessControlService.CAN_READ;
+
+@Service
+@Slf4j
+public class ElasticsearchQueryHelper {
+
+    private static final String SORT = "sort";
+    private static final String CASE_DATA_PREFIX = "data.";
+    private static final String COLLECTION_VALUE_SUFFIX = ".value";
+    private static final String KEYWORD_SUFFIX = ".keyword";
+
+    private final ObjectMapper objectMapper;
+    private final ApplicationParams applicationParams;
+    private final ObjectMapperService objectMapperService;
+    private final SearchQueryOperation searchQueryOperation;
+    private final GetCaseTypeOperation getCaseTypeOperation;
+    private final UserService userService;
+    private final ElasticsearchMappings elasticsearchMappings;
+
+    @Autowired
+    public ElasticsearchQueryHelper(@Qualifier("DefaultObjectMapper") ObjectMapper objectMapper,
+                                    ApplicationParams applicationParams,
+                                    ObjectMapperService objectMapperService,
+                                    SearchQueryOperation searchQueryOperation,
+                                    @Qualifier(AuthorisedGetCaseTypeOperation.QUALIFIER) GetCaseTypeOperation getCaseTypeOperation,
+                                    UserService userService,
+                                    ElasticsearchMappings elasticsearchMappings) {
+        this.objectMapper = objectMapper;
+        this.applicationParams = applicationParams;
+        this.objectMapperService = objectMapperService;
+        this.searchQueryOperation = searchQueryOperation;
+        this.getCaseTypeOperation = getCaseTypeOperation;
+        this.userService = userService;
+        this.elasticsearchMappings = elasticsearchMappings;
+    }
+
+    public CrossCaseTypeSearchRequest prepareRequest(List<String> caseTypeIds, String useCaseString, String jsonSearchRequest) {
+        UseCase useCase;
+        try {
+            useCase = UseCase.valueOfReference(useCaseString);
+        } catch (IllegalArgumentException ex) {
+            throw new BadSearchRequest(String.format("The provided use case '%s' is unsupported.", useCaseString));
+        }
+
+        rejectBlackListedQuery(jsonSearchRequest);
+
+        final List<String> updatedCaseTypeIds = buildCaseTypeIds(caseTypeIds);
+
+        JsonNode searchRequest = stringToJsonNode(jsonSearchRequest);
+        if (useCase != UseCase.DEFAULT) {
```

Comment:
I wouldn't introduce a concept of Default use case. It's a bit misleading we don't really have a default use case. If use case is not passed then is missing. 
Can't we just check whether the string is empty or null?